lib/list.c

```diff
@@ -48,3 +49,33 @@ void add_node_end(struct slist *head, struct slist *node)
 	node->next = NULL;
     }
 }
+
+/* if pos > length of linked list, insert at the end */ 
+void add_node_pos(struct slist **head, struct slist *node, int insert_at_pos)
+{
+    int count;
+    struct slist *pos, *prev_pos;
+    struct slist *headp = *head;
+
+    pos = prev_pos = NULL;
+
+    // fix invalid user input
+    if (insert_at_pos <= 0)
+	insert_at_pos = 1;
+    
+    for (count = 1, pos = headp; pos != NULL; pos = pos->next, count++) {
```

Comment:
Why not loop till "count == insert_at_pos"

for(count = 1, pos = headp; count != insert_at_pos; pos = pos->next, count++);